pkg/service/audit_utils.go

```diff
@@ -389,6 +391,57 @@ func CreateApiKeyAuditEvent(ctx context.Context, al *zap.Logger, action string,
 	}
 }
 
+func GenerateApiKeyAuditEvent(ctx context.Context, al *zap.Logger, action string, id string) {
```

Comment:
I don't think we need this, we have CreateApiKeyAuditEvent which is already used, just the condition needs to be corrected.